src/gsi/combine_radobs.f90

```diff
@@ -50,6 +51,7 @@ subroutine combine_radobs(mype_sub,mype_root,&
   integer(i_kind)                    ,intent(in   ) :: npe_sub,itxmax
   integer(i_kind)                    ,intent(in   ) :: nele
   integer(i_kind)                    ,intent(in   ) :: mpi_comm_sub
+  integer(i_kind)                    ,intent(in   ) :: number_profiles
```

Comment:
It doesn't seem that `number_profiles` is being used.  Was this intended to be used in the `allocate` statement on line 88?

Reply:
@DavidHuber-NOAA I removed the mins in the allocation statement because the code failed at line 98 ( nloc(ndata) =k ).  This suggests ndata gets larger than number_profiles.  Since number_profiles is called ndata in combine_radobs,  It seems sending in number_profiles is a moot point.  

I am going to restore the various read routines and run some cycled tests but leave the changes involving nread in place.

Reply:
@DavidHuber-NOAA I made the various changes with the help of Russ.  My branch is up to date.  My cycled runs on S4 are identical. I'm runn ctests on hera, jet and hercules.
Will resolve this conversation.